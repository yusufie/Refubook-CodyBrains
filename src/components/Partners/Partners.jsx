import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import { useTranslation } from 'react-i18next';
import partners from '../../data/partnersData';

const Partners = () => {
  const { t } = useTranslation();
  return (
    <div
      className="flex flex-col items-center justify-center w-full bg-refubookWhite px-8 py-4"
      dir="ltr"
    >
      <h1 className="text-3xl text-refubookBlack font-bold text-center">
        {t('Our_Partners')}
      </h1>
      <Swiper
        spaceBetween={5}
        pagination={{
          clickable: true,
        }}
        modules={[Pagination]}
        className="w-full "
        loop
        breakpoints={{
          320: {
            slidesPerView: 2,
            spaceBetween: 10,
          },
          640: {
            slidesPerView: 3,
            spaceBetween: 5,
          },
          768: {
            slidesPerView: 4,
            spaceBetween: 5,
          },
          1024: {
            slidesPerView: 5,
            spaceBetween: 5,
          },
          1280: {
            slidesPerView: 6,
            spaceBetween: 5,
          },
        }}
      >
        {partners.map((item) => {
          return (
            <SwiperSlide
              key={item.id}
              className="flex items-center justify-center "
            >
              <img
                src={item.image}
                alt="logo"
                className=" block w-40 h-40 object-contain  "
              />
            </SwiperSlide>
          );
        })}
      </Swiper>
    </div>
  );
};

export default Partners;
