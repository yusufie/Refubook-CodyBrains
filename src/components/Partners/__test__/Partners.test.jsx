import renderer from 'react-test-renderer';
import Partners from '../Partners';

// Mocking Swiper and SwiperSlide to avoid errors in tests

jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));

jest.mock('swiper/react', () => ({
  Swiper: 'Swiper',
  SwiperSlide: 'SwiperSlide',
}));

jest.mock('swiper', () => ({
  Pagination: 'Pagination',
}));

describe('Partners', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Partners />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
