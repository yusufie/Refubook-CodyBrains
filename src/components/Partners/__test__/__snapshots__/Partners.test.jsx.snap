// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners renders correctly 1`] = `
<div
  className="flex flex-col items-center justify-center w-full bg-refubookWhite px-8 py-4"
  dir="ltr"
>
  <h1
    className="text-3xl text-refubookBlack font-bold text-center"
  >
    Our_Partners
  </h1>
  <Swiper
    breakpoints={
      Object {
        "1024": Object {
          "slidesPerView": 5,
          "spaceBetween": 5,
        },
        "1280": Object {
          "slidesPerView": 6,
          "spaceBetween": 5,
        },
        "320": Object {
          "slidesPerView": 2,
          "spaceBetween": 10,
        },
        "640": Object {
          "slidesPerView": 3,
          "spaceBetween": 5,
        },
        "768": Object {
          "slidesPerView": 4,
          "spaceBetween": 5,
        },
      }
    }
    className="w-full "
    loop={true}
    modules={
      Array [
        "Pagination",
      ]
    }
    pagination={
      Object {
        "clickable": true,
      }
    }
    spaceBetween={5}
  >
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="recoded.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="spark.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="flatiron.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="children.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="unhcr.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="amnesty.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="oxfam.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="giz.png"
      />
    </SwiperSlide>
    <SwiperSlide
      className="flex items-center justify-center "
    >
      <img
        alt="logo"
        className=" block w-40 h-40 object-contain  "
        src="unicef.png"
      />
    </SwiperSlide>
  </Swiper>
</div>
`;
